src/coreclr/jit/lsra.cpp

```diff
@@ -8855,6 +8855,13 @@ void LinearScan::handleOutgoingCriticalEdges(BasicBlock* block)
         consumedRegs = switchTable->gtRsvdRegs;
         GenTree* op1 = switchTable->gtGetOp1();
         GenTree* op2 = switchTable->gtGetOp2();
+#if defined(TARGET_ARM64) || defined(TARGET_LOONGARCH64) || defined(TARGET_RISCV64) || defined(TARGET_X86)
+        if (op1->IsLocal())
+        {
+            GenTreeLclVarCommon* lcl = op1->AsLclVarCommon();
+            terminatorNodeLclVarDsc  = &compiler->lvaTable[lcl->GetLclNum()];
+        }
+#endif
```

Comment:
Nit: can we merge it with below if to make it consistent with the `BBJ_COND` case? I.e.
```
if (op1->OperIs(GT_COPY))
{
  // code below
}
else if (op1->IsLocal())
{
  // new code
}
```

Also, I think `op2` should get the handling too

Reply:
> Also, I think op2 should get the handling too

For `GT_SWITCH`, isn't the `op2` always `GT_JMPTABLE`?

Reply:
I think so. But I don't think anything guarantees LSRA doesn't add a `GT_COPY` on top of it, or something in the future doesn't introduce a local... I think just making it consistent with all the other cases doesn't hurt. In fact we could probably unify all handling here by writing the code something like:
```cpp
if (block->HasTerminator())
{
  LIR::AsRange(block).LastNode()->VisitOperands([](GenTree* op) {
    if (op->OperIs(GT_COPY))
    {
        ...
    }
    else if (op->IsLocal())
    {
        ...
    }
    return VisitResult::Continue;
  });
}
```
which would also be a bit more future proof. But I'm also ok with just keeping the special casing for `BBJ_COND`/`BBJ_SWITCH`.